msgType.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+typedef enum msgType	//type of messages used as enum
+{
+	joinRequest=0x01,	//used in pairing mode for join request
+	joinAccept,			//used in pairing mode for join accept
+	joinConfirm,		//used in pairing mode for join confirmation
+	configReq,			//used in configuration mode for requesting configuration message
+	configMsg,			//used in configuration mode for sending configuration message
+	configAck,			//used in configuration mode for acknowledging configuration successfull
+	sensorData,			//used in monitoring mode for sending sensor data and battery level
+	ack,				//used for acknowledging a message received
+	motorStatus,		//used for acknowleging status of motor
+	alarm,				//used to send warning messages/error messages
+	initController		//used to notify level monitor that the controller is recently initiated
+}msgType; 
+
+typedef enum devType {
+	monitor = 0x01,
+	controller,
+	relayUnit
+}devType;
+
+typedef enum response {
+	failure = 0x00,
+	success
+}response;
+
+typedef enum motorState {
+	on = 0x00,
+	off,
+	dryRun
+}motorState;
+/*
+* Message type: Join Request
+* devType: Device type
+*/
+typedef struct joinRequest {
+	unsigned char devType;
+}joinRequest;
+
+/*
+* Message type: Join Accept
+* devID: Device ID, 
+* secretKey: AES Secret Key
+*/
+typedef struct joinAccept {
+	unsigned char devID[4];
+	unsigned char secretKey[16];
+	}joinAccept;
+
+/*
+* Message type: Join Confirm
+* joinAck: Acknowledgement for joining confirmation
+*/
+typedef struct joinConfirm {
```

Comment:
ClientResponse?? One response message is enough for all

Reply:
ok